sdk_extensions/resources/src/main/java/io/opentelemetry/sdk/extensions/resources/ProcessResource.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2020, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.sdk.extensions.resources;
+
+import io.opentelemetry.common.Attributes;
+import io.opentelemetry.sdk.resources.ResourceConstants;
+import io.opentelemetry.sdk.resources.ResourceProvider;
+import java.io.File;
+import java.lang.management.ManagementFactory;
+import java.lang.management.RuntimeMXBean;
+
+/** {@link ResourceProvider} which provides information about the current running process. */
+public class ProcessResource extends ResourceProvider {
+  @Override
+  protected Attributes getAttributes() {
+    Attributes.Builder attributes = Attributes.newBuilder();
+
+    // TODO(anuraaga): Use reflection to get more stable values on Java 9+
+    RuntimeMXBean runtime = ManagementFactory.getRuntimeMXBean();
+    long pid = -1;
+    String runtimeName = ManagementFactory.getRuntimeMXBean().getName();
+    int atIndex = runtimeName.indexOf('@');
+    if (atIndex >= 0) {
+      String pidString = runtimeName.substring(0, atIndex);
+      try {
+        pid = Long.parseLong(pidString);
+      } catch (NumberFormatException ignored) {
+        // Ignore parse failure.
+      }
+    }
+
+    attributes.setAttribute(ResourceConstants.PROCESS_PID, pid);
+
+    String javaHome = null;
+    String osName = null;
+    try {
+      javaHome = System.getProperty("java.home");
+      osName = System.getProperty("os.name");
+    } catch (SecurityException e) {
+      // Ignore
+    }
+    if (javaHome != null) {
+      StringBuilder executablePath = new StringBuilder(javaHome);
+      executablePath
+          .append(File.pathSeparatorChar)
+          .append("bin")
```

Comment:
interesting assumption. This is definitely the most common location for a java executable, but is it universal enough to be encoded here?

Reply:
Let's go with it for now. I'm sure a user on some obscure platform will let us know when it's wrong. :)

Reply:
I wonder what this does on Android, for instance. I also hope no one is running this SDK on android, since we haven't been (and don't intend to be) optimizing for that runtime environment.

Reply:
Currently it's guaranteed to crash because java management isn't available anyways :) I added a note to the README and note to the issue that we can reflectionize this too to unbreak.